org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/IThreadUtility.java

```diff
@@ -0,0 +1,22 @@
+package org.eclipse.jdt.ls.debug;
+
+import com.sun.jdi.ThreadReference;
+import com.sun.jdi.event.StepEvent;
+import java.util.function.Consumer;
+
+import org.eclipse.debug.core.DebugEvent;
+
+interface IThreadUtility {
+    static void stepOver(ThreadReference thread, Consumer<DebugEvent> eventHandler) {
```

Comment:
The real event is inside DebugEvent. DebugEvent is to wrap the original event and provide the handler a way of stop resuming by setting its shouldResume to false.